### PR TITLE
Deprecate `decimal_cols_as_float` in ORC reader (C++ layer)

### PR DESCRIPTION
**Pull request by vuule**
Issue #10129

### CODE REVIEW

**Review by ttnghia — APPROVED** [review]

File: cpp/include/cudf/io/orc.hpp [hunk 1]
```diff
@@ -220,7 +220,9 @@ class orc_reader_options {
    *
    * @param val Vector of fully qualified column names.
    */
-  void set_decimal_cols_as_float(std::vector<std::string> val)
+  [[deprecated(
```
Comment (bdice) [inline cpp/include/cudf/io/orc.hpp:223]: Does this add a compiler warning when building libcudf?

Comment (vuule) [reply cpp/include/cudf/io/orc.hpp:223]: It should issue warnings only when the function is used.
https://en.cppreference.com/w/cpp/language/attributes/deprecated

**Review by vyasr — APPROVED** [review]
Should we also update the documentation to indicate that the method is deprecated? I don't think we have any real policies around this in libcudf, and I doubt many people actually read our doxygen, but it might still be nice to add a message. Otherwise LGTM.

**Comment by vuule** [comment]
> Should we also update the documentation to indicate that the method is deprecated? I don't think we have any real policies around this in libcudf, and I doubt many people actually read our doxygen, but it might still be nice to add a message. Otherwise LGTM.

I haven't done so when previously deprecating APIs, but I'm not opposed to it. Do you know of an example where we included this info in the docs?

**Comment by vyasr** [comment]
> > Should we also update the documentation to indicate that the method is deprecated? I don't think we have any real policies around this in libcudf, and I doubt many people actually read our doxygen, but it might still be nice to add a message. Otherwise LGTM.
> 
> I haven't done so when previously deprecating APIs, but I'm not opposed to it. Do you know of an example where we included this info in the docs?

I do not. It would be nice for us to have some policy around how deprecations should be handled in libcudf (we have a rough one in cuDF Python) but no need to hold up this PR for that if we don't already have a standard. It might be something we could define and document in the dev guide for future use, though.

**Comment by vuule** [comment]
@gpucibot merge

File: cpp/include/cudf/io/orc.hpp (hunk 1 above)
Comment (bdice) [reply cpp/include/cudf/io/orc.hpp:223]: I see. I read that page earlier but thought it was showing compiler output and not runtime output. Thanks for the correction.

**Comment by vyasr** [comment]
@vuule I created https://github.com/rapidsai/cudf/issues/10159 to track this.

**Pull request merged**